block-producer/src/block_builder/mod.rs

```diff
@@ -84,13 +84,20 @@ where
         let updated_accounts: Vec<_> =
             batches.iter().flat_map(TransactionBatch::updated_accounts).collect();
 
-        let created_notes: Vec<(usize, usize, NoteCreated)> = batches
+        let created_notes: Vec<NoteCreated> = batches
             .iter()
             .enumerate()
-            .flat_map(|(batch_idx, batch)| {
-                batch.created_notes().enumerate().map(move |(note_idx_in_batch, note)| {
-                    (batch_idx, note_idx_in_batch, note.clone())
-                })
+            .flat_map(|(batch_index, batch)| {
+                batch.created_note_envelopes_with_details().enumerate().map(
+                    move |(note_index, (note_envelope, details))| NoteCreated {
```

Comment:
Similar conversion is used 2 times in code. Consider writing `From` implementation in proto/domain.

Reply:
Could you point me to the seconde instance?

Reply:
> Could you point me to the seconde instance?

Please, see comments below.